services/system/Accounts/ui/src/components/AccountSelection.tsx

```diff
@@ -0,0 +1,722 @@
+
+  const onSubmit = async (values: z.infer<typeof formSchema>) => {
+    if (isCreatingAccount) {
+      void (await createAccount(values.username));
+    } else {
+      void (await importAccount(values.username));
+    }
+    setIsModalOpen(false);
+  };
+
+  const username = form.watch("username");
+
+  const [debouncedAccount, setDebouncedAccount] = useState<string>();
+
+  const {
+    data: accounts,
+    isLoading: isFetching,
+    error,
+    refetch: fetchAccounts,
+  } = useQuery({
+    queryKey: ["availableAccounts"],
+    queryFn: async (): Promise<AccountType[]> => {
+      supervisor.preLoadPlugins([{ service: "accounts" }]);
+
+      const res = await supervisor.functionCall({
+        method: "getAllAccounts",
+        params: [],
+        service: "accounts",
+        intf: "admin",
+      });
+
+      return z
+        .string()
+        .array()
+        .parse(res)
+        .map((x) => ({ account: x, id: x }));
+    },
+  });
+
+  const isNoAccounts = accounts ? accounts.length == 0 : false;
+
+  console.log({ error }, "was the error");
+
+  const { data: status, isLoading: debounceIsLoading } = useQuery<
+    z.infer<typeof AccountNameStatus>
+  >({
+    queryKey: ["userAccount", debouncedAccount],
+    queryFn: async () => isAccountAvailable(debouncedAccount!),
+    enabled: !!debouncedAccount,
+    initialData: "Loading",
+  });
+
+  const isProcessing = debounceIsLoading || username !== debouncedAccount;
+
+  const accountStatus = isProcessing ? "Loading" : status;
+
+  const isAccountsLoading = isFetching && !accounts;
+  const [selectedAccountId, setSelectedAccountId] = useState<string>();
+
+  useEffect(() => {
+    const debouncedCb = debounce((formValue) => {
+      setDebouncedAccount(formValue.username);
+    }, 1000);
+
+    const subscription = form.watch(debouncedCb);
+
+    return () => subscription.unsubscribe();
+  }, [form]);
+
+  const [activeSearch, setActiveSearch] = useState("");
+
+  const accountsToRender = activeSearch
+    ? (accounts || []).filter((account) =>
+        account.account.toLowerCase().includes(activeSearch.toLowerCase())
+      )
+    : accounts || [];
+  const [searchParams] = useSearchParams();
+
+  const token = searchParams.get("token");
+
+  const selectedAccount = (accountsToRender || []).find(
+    (account) => account.id == selectedAccountId
+  );
+
+  const { data: decodedToken, isLoading: isLoadingToken } =
+    useDecodeToken(token);
+  const isInvite = decodedToken?.tag === "invite-token";
+
+  const {
+    data: invite,
+    isLoading: isLoadingInvite,
+    refetch: refetchToken,
+  } = useDecodeInviteToken(token, decodedToken?.tag == "invite-token");
+
+  const { data: connectionToken, isLoading: isLoadingConnectionToken } =
+    useDecodeConnectionToken(token, decodedToken?.tag == "connection-token");
+
+  const isInitialLoading =
+    isLoadingInvite || isLoadingConnectionToken || isLoadingToken;
+
+  console.log({ decodedToken, inviteToken: invite, connectionToken });
```

Comment:
This also results in a number of console.logs when all three inner properties are undefined